vivid-server/src/modules/friends/friends.service.ts

```diff
@@ -108,15 +108,18 @@ export class FriendsService {
   // deleting the friendship or decline friendrequest
   async deleteFriendship(
     userId: string,
-    friendRequestId: string,
+    friendId: string,
   ): Promise<DeleteResult> {
     return await this.friendsRepository
       .createQueryBuilder()
       .delete()
-      .where('id = :id', { id: friendRequestId })
-      .andWhere('user_1 = :u1 OR user_2 = :u2', {
+      .where('user_1 = :u1 AND user_2 = :u2', {
         u1: userId,
-        u2: userId,
+        u2: friendId,
+      })
+      .orWhere('user_1 = :u3 AND user_2 = :u4', {
+        u3: friendId,
+        u4: userId,
       })
```

Comment:
do a comparison check to see which order they need to be (`user_1 < user_2` is always true)